### PR TITLE
0174 adding guards to collation

### LINKED ISSUE #174: Transaction passes collation, but fails in submission
### Description

User signs a transaction with the required signatures -> collation process is successful -> transaction fails with invalid signatures. 

`2021-12-17 13:06:05.673 [main] ERROR com.hedera.hashgraph.client.cli.ToolsMain.main(ToolsMain.java:60) - Error in executing command: class com.hedera.hashgraph.client.cli.options.SubmitCommand, message: java.util.concurrent.ExecutionException: com.hedera.hashgraph.client.core.exceptions.HederaClientRuntimeException: com.hedera.hashgraph.sdk.PrecheckStatusException: Hedera transaction 0.0.76@1639746360.10000 failed pre-check with the status INVALID_SIGNATURE`

### Steps to reproduce

1. Create a transaction
2. Sign the transaction with valid signatures
3. Submit the transaction to the network

### Additional context

_No response_

### Hedera network

mainnet

### Version

v0.3.0

### Operating system

Other

### PR DESCRIPTION
**Pull request by DSMatusevich**
**Description**:
The app should check if the transactions are the same before collation.

**Related issue(s)**:

Fixes #174 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 package com.hedera.hashgraph.client.cli.options;
 
 import com.google.protobuf.InvalidProtocolBufferException;
-import com.hedera.hashgraph.client.cli.helpers.CollatorHelper;
+import com.hedera.hashgraph.client.core.helpers.CollatorHelper;
```
Comment (MarcKriguerAtHedera) [inline tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java:22]: Nit: This import line should come after the next one (core.helpers > core.action).

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/SubmitCommand.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 package com.hedera.hashgraph.client.cli.options;
 
 import com.google.protobuf.InvalidProtocolBufferException;
-import com.hedera.hashgraph.client.cli.helpers.TransactionCallableWorker;
+import com.hedera.hashgraph.client.core.helpers.TransactionCallableWorker;
```
Comment (MarcKriguerAtHedera) [inline tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/SubmitCommand.java:22]: Same nit here.

File: tools-core/src/main/java/com/hedera/hashgraph/client/core/helpers/TransactionCallableWorker.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package com.hedera.hashgraph.client.cli.helpers;
+package com.hedera.hashgraph.client.core.helpers;
```
Comment (MarcKriguerAtHedera) [inline tools-core/src/main/java/com/hedera/hashgraph/client/core/helpers/TransactionCallableWorker.java:19]: And here -- client.core.action should come before client.core.helpers.

File: tools-core/src/test/java/com/hedera/hashgraph/client/core/helpers/TransactionCallableWorkerTest.java [hunk 4]
```diff
@@ -16,12 +16,13 @@
  * limitations under the License.
  */
 
-package com.hedera.hashgraph.client.cli.helpers;
+package com.hedera.hashgraph.client.core.helpers;
```
Comment (MarcKriguerAtHedera) [inline tools-core/src/test/java/com/hedera/hashgraph/client/core/helpers/TransactionCallableWorkerTest.java:19]: Again here.

**Review by MarcKriguerAtHedera — APPROVED** [review]
Same nit (about order of "import" lines) in 4 different places, due to code package changing in name.  Not an important one, but I believe SonarCloud will call it a code smell if left in the same ordering.

**Comment by DSMatusevich** [comment]
> Same nit (about order of "import" lines) in 4 different places, due to code package changing in name. Not an important one, but I believe SonarCloud will call it a code smell if left in the same ordering.

Thanks for the review Marc. I leave the order of imports to IntelliJ. If you know what setting to change to change the organization, please let me know.

**Pull request merged**